### PR TITLE
Support ssl socket property hostname-verifier

### PR DESCRIPTION
**Pull request by ruhan1**
This depends on https://github.com/Commonjava/jhttpc/pull/36

### CODE REVIEW

File: transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/model/HttpLocation.java [hunk 1]
```diff
@@ -39,4 +39,5 @@
 
     int getProxyPort();
 
+    boolean isHostnameVerifier();
```
Comment (jdcasey) [inline transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/model/HttpLocation.java]: Can we rename this to `isHostnameVerified()`? It'll improve readability / self-documentation.

Comment (ruhan1) [reply transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/model/HttpLocation.java]: Sure. Updated.

**Comment by ruhan1** [comment]
retest this please

**Comment by ruhan1** [comment]
retest this please

**Comment by ruhan1** [comment]
retest this please

**Comment by ruhan1** [comment]
retest this please

**Comment by ruhan1** [comment]
retest this please

**Comment by ruhan1** [comment]
It seems this build does not take the latest jhttpc snapshot. But I can build it successfully on local after cleaning the jhttpc under my ~/.m2/repository/org/commonjava/util/ . This is weird. 
retest this tomorrow

**Comment by ruhan1** [comment]
retest this please

**Comment by ruhan1** [comment]
retest this please

**Pull request merged**